flutter/lib/widgets/google_fab/expandable_fab.dart

```diff
@@ -0,0 +1,145 @@
+//Flutter Packages
+import 'package:flutter/material.dart';
+
+//My Packages
+import 'expanding_action_button.dart';
+
+@immutable
+class ExpandableFab extends StatefulWidget {
+  const ExpandableFab({
+    Key? key,
+    this.initialOpen,
+    required this.distance,
+    required this.children,
+  }) : super(key: key);
+
+  final bool? initialOpen;
+  final double distance;
+  final List<Widget> children;
+
+  @override
+  _ExpandableFabState createState() => _ExpandableFabState();
+}
+
+class _ExpandableFabState extends State<ExpandableFab> with SingleTickerProviderStateMixin{
+  bool _open = false;
+
+  late final AnimationController _controller;
+  late final Animation<double> _expandAnimation;
+  // ...
+
+  @override
+  void initState() {
+    super.initState();
+    _open = widget.initialOpen ?? false;
+    _controller = AnimationController(
+      value: _open ? 1.0 : 0.0,
+      duration: const Duration(milliseconds: 250),
+      vsync: this,
+    );
+    _expandAnimation = CurvedAnimation(
+      curve: Curves.fastOutSlowIn,
+      reverseCurve: Curves.easeOutQuad,
+      parent: _controller,
+    );
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  void _toggle() {
+    setState(() {
+      _open = !_open;
+      if (_open) {
+        _controller.forward();
+      } else {
+        _controller.reverse();
+      }
+    });
+  }
+
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox.expand(
+      child: Stack(
+        alignment: Alignment.bottomRight,
+        clipBehavior: Clip.none,
+        children: [
+          _buildTapToCloseFab(),
+          ..._buildExpandingActionButtons(),
+          _buildTapToOpenFab(),
```

Comment:
**question**: Calling functions that return widgets can be problematic. Can you explain why?

Reply:
Yeah, I agree. Will resolve this.